plugin.video.edltv/default.py

```diff
@@ -0,0 +1,317 @@
+    sql = sql.rstrip(' OR')
+    sql = sql + 'EXCEPT SELECT code, name, parent, location, url FROM edltv where length(code) < 10'
+    rows = c.execute(sql)
+    rows = c.fetchall()
+    return rows
+
+def remove_minus (l) :
+    sql = ''
+    buffer_l = []
+    buffer_l.extend(l)
+    for i in l :
+        if i == '-' :
+            buffer_l.remove(i)
+        else :
+            buffer_l[buffer_l.index(i)] = i.rstrip('-')
+    buffer_l = filter(None, buffer_l)
+    return query_data (buffer_l)
+
+def check_one_word (l) :
+    sql = ''
+    if len(l[0]) > 1 :        
+        if l[0][0] != '-' :
+            sql = 'SELECT code, name, parent, location, url FROM edltv WHERE name like "%'+l[0]+'%" AND filename IS NOT NULL'
+    elif len(l[0]) == 1 :
+        if l[0][0] == '-' or l[0][0] == '+' :
+            sql = 'SELECT code, name, parent, location, url FROM edltv WHERE name like "%'+l[0]+'%" AND filename IS NOT NULL'
+    sql = sql + ' EXCEPT SELECT code, name, parent, location, url FROM edltv where length(code) < 10 ORDER BY code'
+    rows = c.execute(sql)
+    rows = c.fetchall()
+    return rows
+
+def set_word_list (l) :
+    global word_list
+    word_list = l
+
+def get_word_list () :
+    global word_list
+    return word_list
+
+def set_string (s) :
+    global string
+    string = s
+
+def get_string () :
+    global string
+    return string
+
+def replace_and_split (string,symbol='+',to_symbol=' ') :
+    buffer_word_list = []
+    string = string.replace(symbol,to_symbol)
+    buffer_word_list = string.split()
+    set_word_list(buffer_word_list)
+    buffer_word_list = []
+
+def word_function (data) :
+    r = []
+    set_string(data)
+    if len(get_string()) == 1 and get_string()[0] == '+' :
+        replace_and_split (get_string(),' ')
+    else :
+        replace_and_split (get_string())
+        
+    if len(get_word_list()) == 1 :
+        r = check_one_word(get_word_list())
+    elif len(get_word_list()) > 1 :
+        r = remove_minus(get_word_list())
+    if r :
+        set_search_list (r)
+    else :
+        return True
+    return False
+
+def search_function () :
+    kb = xbmc.Keyboard(None,'Search')
+    kb.doModal()
+    if (kb.isConfirmed()):
+        error = word_function (kb.getText())
+        if error :
+            ok = dialog.ok('eDLTV Error', 'ไม่พบข้อมูลที่ต้องการค้นหา')
+
+def router(paramstring):
+    params = dict(parse_qsl(paramstring))
+    if params:
+        if params['action'] == 'listing':
+            if params['category'] == 'ค้นหา':
+                search_function()
+            elif params['hasfolder'] == 'yes':
+                list_categories(params['category'])
+            else :
+                list_videos(params['category'])
+        elif params['action'] == 'play':
+            play_video(params['video'])
+        else:
+           raise ValueError('Invalid paramstring: {0}!'.format(paramstring))
+    else:
+        get_socket_data()
+        set_class()
+
+def get_socket_data () :
```

Comment:
Why do you even need this? This seems to be SSDP broadcasts to discover which UPnP devices are available on the network and later you concatenate a static path to the response (which is not a single url). Was this intentional or did you just grab the code from somewhere else? What is the reason for you to do this?